### PR TITLE
Add sRGB and linear blending example

### PR DESCRIPTION
**Pull request by Zoxc**
This tests if sRGB and linear framebuffers are working as expected.

### CODE REVIEW

File: examples/srgb-blend/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+    }
+
+    fn render(
+        &mut self,
+        view: &wgpu::TextureView,
+        device: &wgpu::Device,
+        queue: &wgpu::Queue,
+        spawner: &wgpu_example::framework::Spawner,
+    ) {
+        device.push_error_scope(wgpu::ErrorFilter::Validation);
```
Comment (cwfitzgerald) [inline examples/srgb-blend/src/main.rs:177]: We don't actually need any manual testing of the error scope - if you don't add an error scope, the default uncaptured error handler panics.

File: examples/common/src/framework.rs [hunk 2]
```diff
@@ -29,6 +29,7 @@ pub enum ShaderStage {
 }
 
 pub trait Example: 'static + Sized {
+    const SRGB: bool = true;
```
Comment (cwfitzgerald) [inline examples/common/src/framework.rs:33]: If we make this a function, there doesn't need to be const generics in the example

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
My only real question is if this should be an example or just a test. I think this is fine as is if you feel strongly, just throwing that possibility out there.

**Comment by Zoxc** [comment]
It's kind of a test, but you need to be able to run it outside the test framework to observe proper surface / presentation behavior.

**Comment by cwfitzgerald** [comment]
Yeah fair enough

File: examples/common/src/framework.rs (hunk 2 above)
Comment (Zoxc) [reply examples/common/src/framework.rs:33]: You'd still need const generics as `framework::run` doesn't take arguments.

**Review by cwfitzgerald — APPROVED** [review]

**Pull request merged**